src/Widgets/DisplayWidget.vala

```diff
@@ -64,4 +64,9 @@ public class Power.Widgets.DisplayWidget : Gtk.Grid {
             return false;
         });
     }
+
+    public void show_percentage (bool show) {
+        percent_revealer.reveal_child = show;
+        percent_revealer.visible = show;
```

Comment:
Just noticed that the visibility of the revealer is controlled by the Indicator - you should not change it here.

Reply:
Any suggestion ?